src/main/java/frc/team2412/robot/commands/shooter/ShooterTargetCommand.java

```diff
@@ -6,49 +6,44 @@
 
 import edu.wpi.first.wpilibj2.command.CommandBase;
 import frc.team2412.robot.subsystem.ShooterSubsystem;
-import frc.team2412.robot.subsystem.ShooterVisionSubsystem;
+import frc.team2412.robot.subsystem.TargetLocalizer;
 import frc.team2412.robot.util.ShooterDataDistancePoint;
 
 public class ShooterTargetCommand extends CommandBase {
     private final ShooterSubsystem shooter;
-    private final ShooterVisionSubsystem vision;
-    private final BooleanSupplier turretEnable;
+    private final TargetLocalizer localizer;
+    private final BooleanSupplier turret;
 
-    public ShooterTargetCommand(ShooterSubsystem shooter, ShooterVisionSubsystem vision) {
+    public ShooterTargetCommand(ShooterSubsystem shooter, TargetLocalizer vision) {
         this(shooter, vision, () -> false);
     }
 
-    public ShooterTargetCommand(ShooterSubsystem shooter, ShooterVisionSubsystem vision, BooleanSupplier turretEnable) {
+    public ShooterTargetCommand(ShooterSubsystem shooter, TargetLocalizer vision, BooleanSupplier turretButton) {
         this.shooter = shooter;
-        this.vision = vision;
-        this.turretEnable = turretEnable;
+        localizer = vision;
+        turret = turretButton;
         addRequirements(shooter);
     }
 
+    double turretAngle = 0;
+
     @Override
     public void execute() {
-        if (!shooter.enableTurret) {
-            return;
-        }
-
-        double distance = 0, yaw = 0;
-
-        if (vision.hasTarget()) {
-            distance = vision.getDistance() + shooter.getDistanceBias();
-            yaw = vision.getYaw() + shooter.getTurretAngleBias();
-        }
 
         if (ShooterConstants.dataPoints != null) {
-            ShooterDataDistancePoint shooterData = ShooterConstants.dataPoints.getInterpolated(distance);
+            ShooterDataDistancePoint shooterData = ShooterConstants.dataPoints
+                    .getInterpolated(localizer.getAdjustedDistance());
             shooter.setHoodAngle(shooterData.getAngle());
             shooter.setFlywheelRPM(shooterData.getRPM());
         }
-
-        if (turretEnable.getAsBoolean()) {
-            shooter.updateTurretAngle(yaw);
+        if (turret.getAsBoolean()) {
+            turretAngle = shooter.getTurretAngle() + localizer.getYaw();
+            shooter.setTurretAngle(turretAngle + localizer.yawAdjustment());
```

Comment:
Because we assign `turretAngle` right before we use it, it's redundant. (Potential changes in the logic would require state between calls, but currently it's unnecessary.)